lib/sandbox/sandbox_test.go

```diff
@@ -0,0 +1,446 @@
+				"imageName", "imageRef", &pb.ContainerMetadata{},
+				"testsandboxid", false, false, false, false, false, "",
+				"/root/for/container", time.Now(), "SIGKILL")
+			testContainer.SetIntermediateMountPoint("/tmp/file")
+			Expect(err).To(BeNil())
+			Expect(testContainer).NotTo(BeNil())
+
+		})
+
+		It("should succeed to add and remove a container", func() {
+			// Given
+			Expect(testSandbox.GetContainer(testContainer.Name())).To(BeNil())
+
+			// When
+			testSandbox.AddContainer(testContainer)
+
+			// Then
+			Expect(testSandbox.GetContainer(testContainer.Name())).
+				To(Equal(testContainer))
+
+			// And When
+			testSandbox.RemoveContainer(testContainer)
+
+			// Then
+			Expect(testSandbox.GetContainer(testContainer.Name())).To(BeNil())
+		})
+
+		It("should succeed to add and remove an infra container", func() {
+			// Given
+			Expect(testSandbox.InfraContainer()).To(BeNil())
+
+			// When
+			err := testSandbox.SetInfraContainer(testContainer)
+
+			// Then
+			Expect(err).To(BeNil())
+			Expect(testSandbox.InfraContainer()).To(Equal(testContainer))
+			Expect(testSandbox.UserNsPath()).NotTo(Equal(""))
+			Expect(testSandbox.NetNsPath()).NotTo(Equal(""))
+
+			// And When
+			testSandbox.RemoveInfraContainer()
+
+			// Then
+			Expect(testSandbox.InfraContainer()).To(BeNil())
+			Expect(testSandbox.UserNsPath()).To(Equal(""))
+			Expect(testSandbox.NetNsPath()).To(Equal(""))
+		})
+
+		It("should fail add an infra container twice", func() {
+			// Given
+			Expect(testSandbox.InfraContainer()).To(BeNil())
+			Expect(testSandbox.SetInfraContainer(testContainer)).To(BeNil())
+
+			// When
+			err := testSandbox.SetInfraContainer(testContainer)
+
+			// Then
+			Expect(err).NotTo(BeNil())
+		})
+
+		It("should fail to set a nil infra container", func() {
+			// Given
+			// When
+			err := testSandbox.SetInfraContainer(nil)
+
+			// Then
+			Expect(err).NotTo(BeNil())
+		})
+	})
+
+	t.Describe("NetNsRemove", func() {
+		It("should succeed when netns nil", func() {
+			// Given
+			// When
+			err := testSandbox.NetNsRemove()
+
+			// Then
+			Expect(err).To(BeNil())
+		})
+	})
+
+	t.Describe("NetNsJoin", func() {
+		It("should fail when network namespace not exists", func() {
+			// Given
+			// When
+			err := testSandbox.NetNsJoin("path", "name")
+
+			// Then
+			Expect(err).NotTo(BeNil())
+		})
+	})
+
+	t.Describe("NetNsCreate", func() {
+		It("should succeed", func() {
+			// Given
+			gomock.InOrder(
+				netNsIfaceMock.EXPECT().Initialized().Return(false),
+				netNsIfaceMock.EXPECT().Initialize().
+					Return(&sandbox.NetNs{}, nil),
+				netNsIfaceMock.EXPECT().SymlinkCreate(gomock.Any()).
+					Return(nil),
+			)
+
+			// When
+			err := testSandbox.NetNsCreate(netNsIfaceMock)
+
+			// Then
+			Expect(err).To(BeNil())
+			Expect(testSandbox.NetNsRemove()).NotTo(BeNil())
+		})
+
+		It("should not crash when parameter is nil", func() {
+			// Given
+			// When
+			_ = testSandbox.NetNsCreate(nil)
+
+			// Then
```

Comment:
Should there be a "expect error to not be nil" here?

Reply:
I removed that check because the `err` result depends on the permissions of the test-running user. It will try to create the path `/var/run/netns` which might be not createable.